### ISSUE TITLE
Connection Error with imap and no log

### ISSUE THREAD

**Issue by lejsue** [Foundry376/Mailspring#1029]
##### Are there any related issues?

No. 

##### What operating system are you using?

macOS High Sierra 10.13.6

##### What version of Mailspring are you using?

1.4.1

--

**Bug?**

When I tried to set up an account to Outlook with imap protocol,  I got "Connection Error - Unable to connect to the server / port you provided. (IMAP)View Log" after I clicked [Connect Account]. 
But when I clicked the "View Log" link, there is no any log in it, only with one line: ----------IMAP----------

I don't have "Show Mailsync Logs" in "Developer" menu, there is "Attach Mailsync to Xcode", and when I click it, I got the error messages on Developer Console:

```
/Applications/Mailspring.app/Contents/Resources/app.asar/src/error-logger.js:101 TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at HTMLBodyElement.WindowEventHandler.AppEnv.commands.add (/src/window-event-handler.es6:110:29)
    at CommandRegistry.dispatch (/src/registries/command-registry.es6:57:35)
    at EventEmitter.WindowEventHandler._electron.ipcRenderer.on.args (/src/window-event-handler.es6:57:23)
    at emitThree (events.js:136:13)
    at EventEmitter.emit (events.js:217:7) {url: "/Applications/Mailspring.app/Contents/Resources/app.asar/src/window-event-handler.js", line: 110, column: 28, pluginIds: Array(0)}

window-event-handler.es6:110 Uncaught TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at HTMLBodyElement.WindowEventHandler.AppEnv.commands.add (/src/window-event-handler.es6:110:29)
    at CommandRegistry.dispatch (/src/registries/command-registry.es6:57:35)
    at EventEmitter.WindowEventHandler._electron.ipcRenderer.on.args (/src/window-event-handler.es6:57:23)
    at emitThree (events.js:136:13)
    at EventEmitter.emit (events.js:217:7)
WindowEventHandler.AppEnv.commands.add @ window-event-handler.es6:110
dispatch @ command-registry.es6:57
WindowEventHandler._electron.ipcRenderer.on.args @ window-event-handler.es6:57
emitThree @ events.js:136
emit @ events.js:217
```

Can someone help me to let me know how to troubleshoot this?
Thanks!

**Comment by ericnewton76** [comment]
I get the same thing.  IMAP and SMTP have been a bit frustrating to setup on a particular mail server run by Kerio.

**Comment by thertz** [comment]
IMAP syncing seems to fail repeatedly on my work account. The headers are downloaded when mailspring is first opened, but they message content is typically not downloaded, and I keep on getting error messages about failure to sync the imap account.

The most annoying part about this, is that the error messages are pop-up messages that halt the program until you hit the OK button. This happens for any message I try to read from my imap account. 

When I open the account preference pane and ask to reconnect, it does reconnect and the issue is 'resolved' for a few minutes, until it shows up again. 

I am attaching my current log file. This has been an ongoing issue, but I thought it had to do with my workplace imap. I worked with the IT folks, there, and tried to access using other clients and everything works well on these clients.

--Tomer
[mailsync-70ec5c49.log](https://github.com/Foundry376/Mailspring/files/2295447/mailsync-70ec5c49.log)

**Comment by ajays-in** [comment]
Facing similar issue with 1.4.2. Imap accounts that were syncing perfectly have started to give Sync errors. This is happening with at-least 2 accounts, and the same accounts are working fine on the phone. 

This looks like a serious regression and makes mailspring pretty much unusable for now

**Comment by lejsue** [comment]
Hi,
I downloaded the source code and traced the logs when calling mailsync:

--------------- Log from buffer ---------------
Waiting for Account JSON:

Waiting for Identity JSON:
info: Identity created at 1534983765 - using ID Schema 1
2018-08-23 17:20:37.249 mailsync[18907:1493370] CFNetwork SSLHandshake failed (-9806)
{"account":null,"error":"ErrorConnection","error_service":"imap","log":"----------IMAP----------\n"}

I guess it is due to my OSX 10.13.6, force CFNetwork to use TLS1.2?

Since mailsync is closed project, I cannot debug it. Please help to figure out this issue. Thanks a lot!

**Comment by gogvale** [comment]
Solved by adding as Office365 account :smiley:

**Comment by chrissipaul** [comment]
I had this issue on on of my accounts, and could only re-authenticate an IMAP account that had ceased to receive messages, once my VPN was turned off.  I have no idea why this would be the case.

**Comment by bengotow** [comment]
Hey folks—thanks for reporting this. The connection / configuration issues with Office365 should be fully resolved now in 1.5.6. Give it a shot and let me know if you still run in to these problems! For what it's worth, Mailspring still doesn't support network proxies. We're hoping to tackle that later this year.

**Comment by kokoshneta** [comment]
I’m not sure if this is related, but I get the same error and empty log when I try setting up IMAP/SMTP accounts hosted on our own domain (using MailEnable on a Windows Server 2012 R2).

I can’t connect to these accounts in most versions of Outlook either (though Mail on macOS works just fine), so there’s probably something additional going on here, but even if I can’t connect, I would still expect there to be some kind of error log actually being output…
